tests/sentry/integrations/bitbucket/test_search.py

```diff
@@ -102,10 +111,13 @@ def test_search_repositories_no_issue_tracker(self, mock_record):
         )
         assert resp.status_code == 400
         assert resp.data == {"detail": "Bitbucket Repository has no issue tracker."}
-        assert len(mock_record.mock_calls) == 4
-        start1, start2, halt1, halt2 = (
-            mock_record.mock_calls
-        )  # calls get, which calls handle_search_issues
+
+        middleware_calls = mock_record.mock_calls[:3] + mock_record.mock_calls[-1:]
+        assert_middleware_metrics(middleware_calls)
+
+        product_calls = mock_record.mock_calls[3:-1]
+
+        start1, start2, halt1, halt2 = product_calls  # calls get, which calls handle_search_issues
```

Comment:
i keep looking at this because i feel like there is a better way but i can't quite seem to come up with anything because everything goes through `EventLifecycleMetric.record_event` 🤔 

i tried mocking the context manager in tests which is where we are starting to record the metric but we're hooking on logic onto that :/